src/lib/project-detection/index.ts

```diff
@@ -44,34 +44,57 @@ export const identifyProjectSlug = async ({
 
 /**
  * Get the pipeline number from the URL
- * @param {string} url - eg: https://app.circleci.com/pipelines/gh/organization/project/2/workflows/abc123de-f456-78gh-90ij-klmnopqrstuv
- * @returns {string} pipeline number - eg: 2
+ * @param {string} url - CircleCI pipeline URL
+ * @returns {number} The pipeline number
+ * @example
+ * // Standard pipeline URL
+ * getPipelineNumberFromURL('https://app.circleci.com/pipelines/gh/organization/project/2/workflows/abc123de-f456-78gh-90ij-klmnopqrstuv')
+ * // returns 2
+ *
+ * @example
+ * // Pipeline URL with complex project path
+ * getPipelineNumberFromURL('https://app.circleci.com/pipelines/circleci/GM1mbrQEWnNbzLKEnotDo4/5gh9pgQgohHwicwomY5nYQ/123/workflows/abc123de-f456-78gh-90ij-klmnopqrstuv')
+ * // returns 123
  */
-export const getPipelineNumberFromURL = (url: string) => {
+export const getPipelineNumberFromURL = (url: string): number => {
   const parts = url.split('/');
-  return parts[7];
-};
+  const pipelineIndex = parts.indexOf('pipelines');
+  if (pipelineIndex === -1) {
+    throw new Error('Invalid CircleCI URL format');
+  }
+  const pipelineNumber = parts[pipelineIndex + 4];
 
-/**
- * Get the project slug from the URL
- * @param {string} url - eg: https://app.circleci.com/pipelines/gh/organization/project/2/workflows/abc123de-f456-78gh-90ij-klmnopqrstuv
- * @returns {string} project slug - eg: gh/organization/project
- */
-export const getProjectSlugFromURL = (url: string) => {
-  const parts = url.split('/');
-  return `${parts[4]}/${parts[5]}/${parts[6]}`;
+  if (!pipelineNumber) {
+    throw new Error('Unable to extract pipeline number from URL');
+  }
+
+  const parsedNumber = Number(pipelineNumber);
+  if (isNaN(parsedNumber)) {
+    throw new Error('Pipeline number in URL is not a valid number');
+  }
+
+  return parsedNumber;
```

Comment:
this function should be a bit more robust now, and always returns a number so we don't have to parse it when called.